👷 Update PR template (#3490)
DataDog/browser-sdk: .github/PULL_REQUEST_TEMPLATE.md

<!-- What does this change exactly? Who will be affected? Include relevant screenshots, videos, links. -->

## Test instructions

<!-- How can the reviewer test this change? Include relevant steps to reproduce the issue, if any. -->

## Checklist

<!-- By submitting this test, you confirm the following: -->

> - [ ] I used AI to generate this PR, please review carefully.

Reviewer: <!-- Thoughts represent an idea that popped up from reviewing. These comments are non-blocking by nature, but they are extremely valuable and can lead to more focused initiatives and mentoring opportunities.  -->
**💭 thought:** ‏ It reads like we're asking the reviewer to do extra work because the author didn't 🤔. Also, We could emphasis self review from the author, even when the PR is not generated by AI.

What about:

```suggestion
- [ ] AI generated.
- [ ] Self reviewed
```

> - [ ] I used AI to generate this PR, please review carefully.
> - [ ] I tested on my local environment.
> - [ ] I deployed and tested on staging.
> - [ ] I wrote unit tests for this change.
> - [ ] I wrote integration tests for this change.

Reviewer 2: <!-- Nitpicks are small, trivial, but necessary changes. Distinguishing nitpick comments significantly helps direct the reader's attention to comments requiring more involvement.  -->
**🥜 nitpick:**  A slightly more concise version:‏
```suggestion
 - [ ] Used AI to generate this PR, please review carefully.
 - [ ] Tested locally.
 - [ ] Deployed and tested on staging.
 - [ ] Added unit tests for this change.
 - [ ] Added integration tests for this change.
```
